### ISSUE TITLE
Sonos binding: invalid subscription ID

### ISSUE THREAD

**Issue by lolodomo** [eclipse/smarthome#1903]
It is not the case with beta 3 but in recent snapshots, we can find in the logs error messages of this kind:
```
`19:15:25.671 [ERROR] [g.jupnp.protocol.sync.ReceivingEvent] - Invalid subscription ID, no active subscription: (IncomingEventRequestMessage) NOTIFY /upnpcallback/dev/RINCON_000E58D01E9001400/svc/upnp-org/DeviceProperties/event/cb SEQUENCE: 0
19:15:25.808 [ERROR] [g.jupnp.protocol.sync.ReceivingEvent] - Invalid subscription ID, no active subscription: (IncomingEventRequestMessage) NOTIFY /upnpcallback/dev/RINCON_000E58D01E9001400_MR/svc/upnp-org/RenderingControl/event/cb SEQUENCE: 0
19:15:25.884 [ERROR] [g.jupnp.protocol.sync.ReceivingEvent] - Invalid subscription ID, no active subscription: (IncomingEventRequestMessage) NOTIFY /upnpcallback/dev/RINCON_000E58D01E9001400/svc/upnp-org/GroupManagement/event/cb SEQUENCE: 0
```
Subscriptions with the Sonos is not valid and as a result channels are not refreshed.

**Comment by kgoderis** [comment]
I presume you get this one after you restart the binding? Then it is normal and these can be neglected

**Comment by lolodomo** [comment]
Note that these errors are noit always present at every startup.

When the erros are present, it looks like the registration is not done, and the channels are not updated.
Maybe the error that is due to an old registration breaks something in the new registration ?

**Comment by lolodomo** [comment]
It looks it happens only if you restart OH quickly. It makes sense that if you wait a little the Sonos should probably stop using the old registration.
But unfortunately, each time it happens, registration by the binding is not active at all.

**Comment by kgoderis** [comment]
Hmm seems like a shortcoming in the UPNP stack in the Sonos then. Maybe we should investigate if we could pick up the subscription information and explicitly unsubscribe these subscriptions. (It has been a long time since I was in that part of the code, i am not sure this is possible at all)

**Comment by lolodomo** [comment]
As you previously said, it should not be a problem to receive data from the Sonos for something now unregistered, we should just ignore it. Maybe a warning would be better than an error in this case.
But by the way, we should initiate a valid registration in all cases.

**Comment by kgoderis** [comment]
The data is ignored, the messages are coming from the jupnp library

**Comment by lolodomo** [comment]
I have in memory that Sonos (bug in firmware ?) does not allow having several subscritptors from the same host. In this case, it only dispatches events to one of them.
In case it could help...

**Comment by lolodomo** [comment]
Will see if a fix for issue #1901 fixes this one too. But that is probable.

**Comment by lolodomo** [comment]
I can see with logs that even when I got invalid subscription ID, the binding tries to subscribe to events.
The only problem is GENA subscriptions that regularly failed at startup and that is covered by issue #1901.
